plugins/inputs/redis_sentinel/redis_sentinel.go

```diff
@@ -0,0 +1,488 @@
+
+				sentinelMastersTags, sentinelMastersFields := convertSentinelMastersOutput(client.BaseTags(), m, quorumErr)
+				acc.AddFields(measurementMasters, sentinelMastersFields, sentinelMastersTags)
+
+				// ------------------------------------------------------------
+
+				// Check other Sentinels
+				sentinelsCmd := redis.NewSliceCmd("sentinel", "sentinels", m["name"])
+				if ssErr := client.sentinel.Process(sentinelsCmd); ssErr != nil {
+					acc.AddError(ssErr)
+					return
+				}
+
+				sentinels, sentinelsErr := sentinelsCmd.Result()
+				if sentinelsErr != nil {
+					acc.AddError(sentinelsErr)
+					return
+				}
+
+				for _, sentinel := range sentinels {
+					sm := toMap(sentinel.([]interface{}))
+
+					sentinelTags, sentinelFields := convertSentinelSentinelsOutput(client.BaseTags(), m["name"], sm)
+					acc.AddFields(measurementSentinels, sentinelFields, sentinelTags)
+				}
+
+				// ------------------------------------------------------------
+
+				// Check other Replicas
+				replicasCmd := redis.NewSliceCmd("sentinel", "replicas", m["name"])
+				if srErr := client.sentinel.Process(replicasCmd); srErr != nil {
+					acc.AddError(srErr)
+					return
+				}
+
+				replicas, replicasErr := replicasCmd.Result()
+				if replicasErr != nil {
+					acc.AddError(replicasErr)
+					return
+				}
+
+				for _, replica := range replicas {
+					rm := toMap(replica.([]interface{}))
+
+					replicaTags, replicaFields := convertSentinelReplicaOutput(client.BaseTags(), m["name"], rm)
+					acc.AddFields(measurementReplicas, replicaFields, replicaTags)
+				}
+
+				// ------------------------------------------------------------
+
+				// Get INFO
+				infoCmd := redis.NewStringCmd("info", "all")
+				if iErr := client.sentinel.Process(infoCmd); iErr != nil {
+					acc.AddError(iErr)
+					return
+				}
+
+				info, infoErr := infoCmd.Result()
+				if infoErr != nil {
+					acc.AddError(infoErr)
+					return
+				}
+
+				rdr := strings.NewReader(info)
+				infoTags, infoFields := convertSentinelInfoOutput(acc, client.BaseTags(), rdr)
+
+				acc.AddFields(measurementSentinel, infoFields, infoTags)
+			}
+		}(client, acc)
+	}
+
+	wg.Wait()
+
+	return nil
+}
+
+// converts `sentinel masters <name>` output to tags and fields
+func convertSentinelMastersOutput(
+	global_tags map[string]string,
+	master map[string]string,
+	quorumErr error,
+) (map[string]string, map[string]interface{}) {
+
+	tags := make(map[string]string)
+	for k, v := range global_tags {
+		tags[k] = v
+	}
+
+	tags["master_name"] = master["name"]
+
+	fields := make(map[string]interface{})
+
+	fields["has_quorum"] = 0
+	if quorumErr == nil {
+		fields["has_quorum"] = 1
+	}
+
+	for key, val := range master {
+		key = strings.Replace(key, "-", "_", -1)
+
+		if ival, err := strconv.ParseInt(val, 10, 64); err == nil {
+			fields[key] = ival
+			continue
+		}
+
+		if fval, err := strconv.ParseFloat(val, 64); err == nil {
+			fields[key] = fval
+			continue
+		}
```

Comment:
We use a strategy like this in a number of places in Telegraf and it often causes issues, since it is possible for a value to change type depending on how large it is.  What would be more robust is to have a map of expected variables and the type they should be parsed as.